kod/object/item/passitem/ring/ringinv.kod

```diff
@@ -62,7 +63,12 @@ messages:
 	      return False;
       }
 
-      propagate;
+      if send(poOwner,@GetKarma) > -50  
+      { 
+         send(poOwner,@MsgSendUser,#message_rsc=ringinvisibility_cant);
```

Comment:
All kod messages need to end by returning something (an empty return statement returns $) or by propagating to a parent class containing a message of the same name (this is the purpose of the propagate statement). Diggie has added the new check in place of the propagate statement, rather than above it (so needs to add the propagate before the end of the message).

ReqUse means RequestUse, it is called if an object is trying to use something. It returns FALSE if the item can't be used, and in the case of the invis ring if there is nothing in that class preventing its use the message will propagate to Ring or to Item where those ReqUse messages will return either FALSE or TRUE. In general the ReqX messages are called before something is attempted.